guides/templates_ecr.md

```diff
@@ -0,0 +1,190 @@
+    import { AccountPrincipal } from '@aws-cdk/aws-iam';
+
+In this file, you set the environment configuration for each galaxy and then instantiate a galaxy per environment. Examples are included in the blank template. Duplicate these and replace the names as needed. It is essential that names match those set in your core CDK-Cosmos, e.g. `Dev` must be consistently named.
+
+    const devEnvConfig = { account: '2222', region: 'ap-southeast-2' };
+    const devGalaxy = new AppGalaxyStack(cosmos, 'Dev', {
+      env: devEnvConfig,
+    });
+
+For each galaxy, add permissions to allow accessing the ECR repo (repo located in the management account)
+
+    // Allow the Dev Galaxy to access the ecr repo
+    cosmos.ecrRepo.grantPull(new AccountPrincipal(devGalaxy.account));
+
+For each solar system, create a new solar system stack, ensuring the name matches with your core Cosmos. This allows the portal construct to use resources from core cosmos. You will also need to pass app version as a prop.
+
+    // // Extend the Dev SolarSystem, by creating service
+    // const dev = new AppSolarSystemStack(devGalaxy, 'Dev', {
+    //   appVersion: process.env.APP_BUILD_VERSION,
+    // });
+
+ > __Important note with solar systems:__ Currently, solar systems must be commented out in the first run of the bootstrapper. You can then uncomment solar systems, push up your changes and re-run the cdk pipeline. See [Getting Started - Extension # Solar Systems](getting_started_extension#solar-systems) for more information. 
+
+
+Optionally, add a deployment stage in the pipeline to target specific solar systems (can pass one or multiple stacks). If not, all stacks will be deployed in the final deploy stage of the pipeline. It is, however, recommended to separate dev and prod as a minimum. 
+
+    // // Add a Deployment stage in App Pipeline to target this
+    // ciCd.addCdkDeployEnvStageToCodePipeline({
+    //   name: 'DeployDev',
+    //   stacks: [dev],
+    //   isManualApprovalRequired: false,
+    // });
+
+Again optionally, you can also add a separate stage in the CDK pipeline to target specific stacks
+
+    // ciCd.addCdkDeployEnvStageToCdkPipeline({
+    //   name: 'DeployDev',
+    //   stacks: [dev],
+    //   isManualApprovalRequired: false,
+    // });
+
+As with the solar system, this should be commented out in the first run (as the solar system targeted by this pipeline stage has not yet been created). When you re-run the cdk pipeline, it will change itself to add the new stage. Changes to the pipeline will make it pause, and you will have to re-run it again. This is a property of CodePipeline, and something that may be addressed in future versions of CDK-Cosmos by implementing the new CDK Pipeline construct provided by AWS. In the meantime, the blank template includes an optional CiCd stage in the pipeline, which will allow you to deploy any changes needed to the CiCd infrastructure first and minimise waiting time on additional runs of the pipeline.
```

Comment:
actually not positive this can't be deployed at the same time as the solarsystem it targets - but I could very well be wrong

Reply:
it can be deployed at the same time as the solar system, just can't be there before the solar system is, ie in the very first run of the pipeline. I'll try to explain that better